src/adam_bashforth_adaptive.jl

```diff
@@ -0,0 +1,371 @@
+      Φ[1,1:min(max(n-1,1),currentStepOrder+1)] = Φ[2,1:min(max(n-1,1),currentStepOrder+1)]
+
+      ## n is incremented
+      n= n+1
+
+    end #while loop over steps
+
+    return vcat(t), y
+end
+
+
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++#
+#HELPER FUNCTIONS FOR ADAPTIVE AB SOLVERS
+###############################################################################
+#Gamma star function from Hairer et al pg 359, table 1.2
+#-used in adaptive step selection (See Harier pg 423)
+###############################################################################
+function γ_star(j)
+    if j == 0
+        return 1
+    elseif j > 0
+
+    return polyval((1 / factorial(j))*polyint(poly(diagm(collect(1-(j-1):1)))),1)
+    else
+        error("order must be positive integer")
+    end
+end
+
+################################################################################
+#Helper functions for calculating coefficients from recurence relationships
+#
+################################################################################
+#Based on Harier Lemma 5.1 on page 399
+#Calculates c_j,q and g_n,j for n and j =1:steporder
+function g_coefs(g,c,h,t,n,steporder)
+    for j = 1:steporder
+      for q = 1:(steporder)-(j-1)
+        if j ==1
+          c[j,q]=1/q
+        elseif j==2
+          c[j,q]=1/q/(q+1)
+        else
+          c[j,q] = c[j-1,q] - c[j-1,q+1]*h[n]/(t[n+1]-t[n-(j-1)+1])
+        end
+      end
+      g[j]=c[j,1]
+    end
+  return g
+end
+
+#Based on Harier equation 5.9. Calculates b, Φ, Φstar for
+#n and j  in range_of_indices
+function Φ_and_Φstar_coefs(b, Φ, Φstar,dy_n,h,t,n,range_of_indices)
+    for j in range_of_indices #0:k-1
+      if j== 1#0
+        b[j] = 1
+        Φ[2,j] = dy_n
+        Φstar[2,j] = dy_n
+      else j>1#0
+        #Note: since n>=j, when this runs, n>=2 (so n-1>=1) and
+        b[j] = b[j-1]*(t[n+1]-t[n-(j-1)+1])/(t[n]-t[n-(j-1)])#!!! Need to fix this
+        Φ[2,j] =  Φ[2,j-1]- Φstar[1,j-1]
+        Φstar[2,j] = b[j]*Φ[2,j]
+      end
+    end
+    return b, Φ, Φstar
+end
+
+
+#Based on Harier equation 5.9. Just calculates the Φ coefficients
+function Φ_coefs(b, Φ, Φstar,dy_n,h,t,n,range_of_indices)
+    for j  in range_of_indices
+      if j== 1#0
+        Φ[3,j] = dy_n
+      else j>1#0
+        #Note: since n>=j, when this runs, n>=2 (so n-1>=1) and
+        Φ[3,j] =  Φ[3,j-1]- Φstar[2,j-1]
+      end
+    end
+    return Φ
+end
+
+################################################################################
+#Helper functions initializing/resizing arrays
+################################################################################
+function intialize_coef_arrays(order,T)
+    c = zeros(order,order)
+    g = zeros(order)
+    b = zeros(order)
+    Φstar = Array{T,2}(2,order)
+    Φ = Array{T,2}(3,order)
+    return c,g,b,Φstar, Φ
+end
+
+function resize_coef_arrays(g,c,b,Φstar,Φ,T,old_steporder,new_steporder)
```

Comment:
Why don't you have a maximal allowed order and then make the coefficient arrays to that size?  Resizing them seems inefficient.